context.go

```diff
@@ -21,25 +21,61 @@
 package fx
 
 import (
-	gcontext "context"
+	"context"
 
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
+	"go.uber.org/fx/service"
 	"go.uber.org/fx/ulog"
 )
 
-// Context embeds Host and Go stdlib context for use
-type Context interface {
-	gcontext.Context
+type contextKey int
 
-	Logger() ulog.Log
+const _contextStore contextKey = iota
+
+type ctxStore struct {
```

Comment:
We had Host in the context earlier, but the object was too big and we decided not to keep it as a whole in the context.